psl/psl-core/src/js_connector.rs

```diff
@@ -1,125 +1,8 @@
-
-    fn provider_name(&self) -> &'static str {
-        self.provider_name
-    }
-
-    fn name(&self) -> &str {
-        self.name
-    }
-
-    fn capabilities(&self) -> ConnectorCapabilities {
-        self.canonical_connector.capabilities()
-    }
-
-    fn max_identifier_length(&self) -> usize {
-        self.canonical_connector.max_identifier_length()
-    }
-
-    fn referential_actions(&self) -> enumflags2::BitFlags<parser_database::ReferentialAction> {
-        self.canonical_connector.referential_actions()
-    }
-
-    fn available_native_type_constructors(&self) -> &'static [NativeTypeConstructor] {
-        self.canonical_connector.available_native_type_constructors()
-    }
-
-    fn scalar_type_for_native_type(&self, native_type: &NativeTypeInstance) -> parser_database::ScalarType {
-        self.canonical_connector.scalar_type_for_native_type(native_type)
-    }
-
-    fn default_native_type_for_scalar_type(&self, scalar_type: &parser_database::ScalarType) -> NativeTypeInstance {
-        self.canonical_connector
-            .default_native_type_for_scalar_type(scalar_type)
-    }
-
-    fn native_type_is_default_for_scalar_type(
-        &self,
-        native_type: &NativeTypeInstance,
-        scalar_type: &parser_database::ScalarType,
-    ) -> bool {
-        self.canonical_connector
-            .native_type_is_default_for_scalar_type(native_type, scalar_type)
-    }
-
-    fn native_type_to_parts(&self, native_type: &NativeTypeInstance) -> (&'static str, Vec<String>) {
-        self.canonical_connector.native_type_to_parts(native_type)
-    }
-
-    fn parse_native_type(
-        &self,
-        name: &str,
-        args: &[String],
-        span: diagnostics::Span,
-        diagnostics: &mut diagnostics::Diagnostics,
-    ) -> Option<NativeTypeInstance> {
-        self.canonical_connector
-            .parse_native_type(name, args, span, diagnostics)
-    }
-
-    fn validate_url(&self, url: &str) -> Result<(), String> {
-        if let Some(allowed_protocols) = self.allowed_protocols {
-            let scheme = url.split(':').next().unwrap_or("");
-            if allowed_protocols.contains(&scheme) {
-                Ok(())
-            } else {
-                Err(format!(
-                    "The URL scheme `{}` is not valid for the {} connector. The following schemes are allowed: {}",
-                    scheme,
-                    self.name,
-                    allowed_protocols.join(", ")
-                ))
-            }
-        } else {
-            self.canonical_connector.validate_url(url)
-        }
-    }
-
-    fn default_relation_mode(&self) -> RelationMode {
-        self.canonical_connector.default_relation_mode()
-    }
-
-    fn allowed_relation_mode_settings(&self) -> BitFlags<RelationMode> {
-        self.canonical_connector.allowed_relation_mode_settings()
-    }
-
-    fn flavour(&self) -> Flavour {
-        self.flavour
-    }
+    /// JS drivers are used in Query Engine.
+    Js,
 }
```

Comment:
Could this be moved to the QE? It's not relevant to the PSL.

Reply:
Sure, any preferred location in particular?

Reply:
nope, anywhere that feels good is fine :+1: